SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/ReferenceCountedLockingPolicy.cpp

```diff
@@ -11,28 +11,19 @@
 
 DEFINE_LOG_CATEGORY(LogReferenceCountedLockingPolicy);
 
-ActorLockToken UReferenceCountedLockingPolicy::AcquireLock(const AActor* Actor, FString DebugString)
+bool UReferenceCountedLockingPolicy::CanAcquireLock(AActor* Actor) const
 {
-	check(Actor != nullptr);
-	if (!CanAcquireLock(Actor)) {
-		UE_LOG(LogReferenceCountedLockingPolicy, Error, TEXT("Called AcquireLock when CanAcquireLock returned false. Actor: %s."), *Actor->GetName());
-		return SpatialConstants::INVALID_ACTOR_LOCK_TOKEN;
-	}
-	++ActorToReferenceCount.FindOrAdd(Actor);
-	UE_LOG(LogReferenceCountedLockingPolicy, Log, TEXT("Acquiring migration lock. "
-		"Actor: %s. Lock name: %s. Token %d: Locks held: %d."), *Actor->GetName(), *DebugString, NextToken, *ActorToReferenceCount.Find(Actor));
-	TokenToNameAndActor.Emplace(NextToken, LockNameAndActor{ MoveTemp(DebugString), Actor });
-	return NextToken++;
-}
+	const USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(Actor->GetWorld()->GetNetDriver());
+	const Worker_EntityId EntityId = NetDriver->PackageMap->GetEntityIdFromObject(Actor);
 
-bool UReferenceCountedLockingPolicy::CanAcquireLock(const AActor* Actor) const
-{
-	check(Actor != nullptr);
-	const auto* NetDriver = Cast<USpatialNetDriver>(Actor->GetWorld()->GetNetDriver());
-	const auto EntityId = NetDriver->PackageMap->GetEntityIdFromObject(Actor);
+	if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
+	{
+		UE_LOG(LogReferenceCountedLockingPolicy, Error, TEXT("CanAcquireLock called for actor without corresponding entity ID. Actor: %s"), *Actor->GetName());
```

Comment:
If this isn't a user facing function anymore then I'd not reference the function name in the log.

Reply:
how bout
```suggestion
		UE_LOG(LogReferenceCountedLockingPolicy, Error, TEXT("Failed to lock actor without corresponding entity ID. Actor: %s"), *Actor->GetName());
```